### ISSUE TITLE
보고서작성할게요

### ISSUE THREAD

**Issue by Lay4U** [Lay4U/AICNN#5]
get model 부에서 Conv2D에 32와 2,2를 왜 주었는지
그밑에 64로 왜쌓았는지,
Dropout을 왜 쌓았는지
Dense의 크기를 왜 1024로 했는지
하이퍼파라미터를 왜 이렇게 설계했는지등의 결과 리포트 써서 깃헙에 올려주세요
커밋 하는 방법 모르시면 카톡으로 보내주세요

제가 취합해서 보고서 내겠습니다.

**Comment by cherub0522** [comment]
강영우 올림

- get model 부에서 Conv2D에 32와 2,2를 왜 주었는지

sample로 준 
model.add(Conv2D(24, (4, 4), padding='same', 
	input_shape=self.x_train.shape[1:], activation='relu'))
에서 밑의 

model.add(Conv2D(32, (2, 2), padding='same', 
	input_shape=self.x_train.shape[1:], activation='relu'))
으로 수정한 이유는 
 
Conv2D()의 컨볼루션 필터의 수를 32로 늘려 출력되는 수를 늘리고, 
컨볼루션 커널의 (행, 열)을 (2,2)로 줄여 영상에 대한 
더 높은 정확도(Accuracy)를 얻기 위해 수정했다. 


 - 그밑에 64로 왜쌓았는지,

sample로 준 
model.add(Conv2D(24, (4, 4), padding='same', 
	input_shape=self.x_train.shape[1:], activation='relu'))
에서 밑의 

model.add(Conv2D(64, (2, 2), activation='relu'))로 변경한 이유는 
Conv2D()의 컨볼루션 필터의 수를 64로 늘려 출력되는 수를 늘리고, 
컨볼루션 커널의 (행, 열)을 (2,2)로 줄여 영상에 대한 
더 높은 정확도(Accuracy)를 얻기 위해 수정했다. 

위와 필터의 수(32)와 다른 이유는 이렇게 하서 더 높은 정확도를 얻었기 때문이다. 

 - Dropout을 왜 쌓았는지 : 

Dropout은 과적합을 방지하기 위해서 학습 시에 지정된 비율만큼 
임의의 입력 뉴런을 제외시키는 기능이 있다. 
그래서 Dropout을 여러 번 쌓아서 Conv2D(32, (2, 2),~ ), Conv2D(64, (2, 2) ~ )으로 
많이 생성된 입력값의 수를 줄이기 위해 Dropout을 쌓았다. 

sample로 준 
MaxPooling2D(pool_size=(3, 3)) 을 
MaxPooling2D(pool_size=(2, 2)) 로 바꾼 이유는 
영상에서의 사소한 변화가 주는 영향을 더 줄이기 위해 pool_size를 축소했다.

참고 : https://tykimos.github.io/DeepBrick/ 
https://tykimos.github.io/2017/01/27/MLP_Layer_Talk/


 - Dense의 크기를 왜 1024로 했는지 : 

Dense는 모든 입력 뉴런과 출력 뉴런을 연결하는 전결합층이다. 
이때 가중치가 높을수록 해당 입력 뉴런이 출력 뉴런에 미치는 영향이 크고, 
낮을수록 미치는 영향이 적다.
가중치는 입력 뉴런과 출력 뉴런의 연결강도이다.
또는 가중치는 입력값 * 출력값이라 볼 수 있다. 
그래서 Dense의 출력 뉴런의 수를 1024로 늘린 이유는 
출력 뉴런의 수를 크게 늘러 가중치를 크게 하여 
입력 뉴런이 출력 뉴런에 큰 영향을 줄 수 있게 하기 위해서이다. 


 - 하이퍼파라미터를 이렇게 설계한 이유

hyper['batch_size'] = 32  # 배치 사이즈
hyper['epochs'] = 20  # epochs은 최대 20 설정 !!
# hyper['learning_rate'] = 0.01  # 학습률
hyper['learning_rate'] = 1  # 학습률

하이퍼파라미터를 이렇게 설계한 이유는 이 설계를 통하여 
가장 적합한 Accuracy를 얻었기 때문이다.

**Comment by Lay4U** [comment]
제가 제시한거는 한예시일 뿐입니다. 인공지능 팀프로젝트.pdf 에 나와있는 평가사항과 보고서를 기준으로 왜 이 모델을 선택했는지, 결과에대한 주관적인 평가 개선할수 있는 방안 등도 넣어주시면 감사하겠습니다.
